internal/db/kvs/redis/option.go

```diff
@@ -350,3 +361,17 @@ func WithInitialPingDuration(dur string) Option {
 		return nil
 	}
 }
+
+// WithHooks returns the option to add hooks
+func WithHooks(hooks ...Hook) Option {
+	return func(r *redisClient) error {
+		if r.hooks != nil {
+			r.hooks = append(r.hooks, hooks...)
+			return nil
+		}
+
+		r.hooks = hooks
```

Comment:
maybe we don't need this? it will set `r.hooks` to nil if `hooks` is nil

Reply:
I think it's ok to overwrite it with nil.
Do we have a coding guide about Option functions like that?

Reply:
I don't think we have a coding guideline for functional option... :| 
if `r.hooks` have already inserted some hooks and when the pass in `hooks` is nil, the `r.hooks` will be overwrited.. 
Can you confirm if it is the correct implementation?

Reply:
ah, i see.
Sorry i'll fix it. :bow: :bow: :bow:

Reply:
@kevindiu i fixed it. Thanks :+1: